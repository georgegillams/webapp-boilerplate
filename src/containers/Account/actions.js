import { createRoutine } from 'redux-saga-routines';

import { LOAD_SHOWCASES } from './constants';

export const loadShowcases = createRoutine(LOAD_SHOWCASES);
