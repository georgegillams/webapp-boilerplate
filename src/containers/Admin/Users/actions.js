import { createRoutine } from 'redux-saga-routines';

import { LOAD, REMOVE, REQUEST_MAGIC_LINK } from './constants';

export const load = createRoutine(LOAD);
export const remove = createRoutine(REMOVE);
export const requestMagicLink = createRoutine(REQUEST_MAGIC_LINK);
