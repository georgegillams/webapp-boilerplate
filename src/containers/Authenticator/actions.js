import { createRoutine } from 'redux-saga-routines';

import { LOAD_AUTH, SET_COOKIES_ALLOWED } from './constants';

export const loadAuth = createRoutine(LOAD_AUTH);
export const setCookiesAllowed = createRoutine(SET_COOKIES_ALLOWED);
