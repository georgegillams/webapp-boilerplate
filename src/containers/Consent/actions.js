import { createRoutine } from 'redux-saga-routines';

import { CONSENT, DEFER_CONSENT, RE_PROMPT_CONSENT, RESET_CONSENT } from './constants';

export const consent = createRoutine(CONSENT);
export const deferConsent = createRoutine(DEFER_CONSENT);
export const rePromptConsent = createRoutine(RE_PROMPT_CONSENT);
export const resetConsent = createRoutine(RESET_CONSENT);
