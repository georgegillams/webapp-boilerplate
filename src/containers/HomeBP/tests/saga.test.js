import { put, takeLatest } from 'redux-saga/effects';

import { loadShowcases } from '../actions';

import dataShowcases, { loadShowcasesRequest } from '../saga';

describe('loadShowcasesRequest Saga', () => {
  let mainSaga;

  beforeEach(() => {
    mainSaga = dataShowcases();
  });

  it('Should get list of showcases TRIGGER', () => {
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(loadShowcases.TRIGGER, loadShowcasesRequest));
  });

  describe('loadShowcases actions', () => {
    let loadShowcasesGenerate;

    const response = [
      {
        title: 'First title',
      },
      {
        title: 'Second title',
      },
    ];

    beforeEach(() => {
      loadShowcasesGenerate = loadShowcasesRequest();

      const selectDescriptor = loadShowcasesGenerate.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should loadShowcasesRequest be success', () => {
      loadShowcasesGenerate.next(response);
      const putShowcasesSuccess = loadShowcasesGenerate.next(response).value;
      loadShowcasesGenerate.next(response);

      expect(putShowcasesSuccess).toEqual(put(loadShowcases.success(response)));
    });

    it('Should loadShowcasesRequest be failure', () => {
      const response = new Error('Some error');
      const putShowcasesFailure = loadShowcasesGenerate.throw(response).value;

      expect(putShowcasesFailure).toEqual(put(loadShowcases.failure(response)));
    });
  });
});
