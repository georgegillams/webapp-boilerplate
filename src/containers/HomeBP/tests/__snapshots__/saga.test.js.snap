// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadShowcasesRequest Saga loadShowcases actions Should loadShowcasesRequest be failure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/showcases/LOAD_SHOWCASES/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`loadShowcasesRequest Saga loadShowcases actions Should loadShowcasesRequest be success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/showcases/LOAD_SHOWCASES/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
